File: business-management-software-frontend/src/services/OrderService.js
import axios from "axios";

const ORDER_API_BASE_URL = "http://localhost:8080/api";

class OrderService {
  createOrder(order) {
    return axios.post(ORDER_API_BASE_URL + "/orders", order);
  }
  getOrders() {
    return axios.get(ORDER_API_BASE_URL + "/orders");
  }
  searchOrders(searchTerm) {
    return axios.get(ORDER_API_BASE_URL + "/orders/search/" + searchTerm);
  }

  getOrderById(orderId) {
    return axios.get(ORDER_API_BASE_URL + "/order/" + orderId);
  }
  updateOrder(order, orderId) {
    console.log("api call");
    return axios.put(ORDER_API_BASE_URL + "/order/" + orderId, order);
  }

  deleteOrder(orderId) {
    return axios.delete(ORDER_API_BASE_URL + "/order/" + orderId);
  }
  getOrderItems(orderId) {
    return axios.get(ORDER_API_BASE_URL + "/order/" + orderId + "/items");
  }
}
export default new OrderService();

File: BusinessManagementSoftwareBackend/src/main/java/com/ali/BusinessManagementSoftwareBackend/services/orders/OrderService.java
package com.ali.BusinessManagementSoftwareBackend.services.orders;

import com.ali.BusinessManagementSoftwareBackend.dto.OrderDto;
import com.ali.BusinessManagementSoftwareBackend.dto.OrderItemDto;
import com.ali.BusinessManagementSoftwareBackend.dto.ProductDto;

import java.io.IOException;
import java.util.List;

public interface OrderService {
    public boolean createOrder(OrderDto orderDto) throws IOException ;
    public List<OrderDto> getOrders();
    public void deleteOrder(Long id);
    public OrderDto getOrderById(Long id);
    public boolean updateOrder(Long orderId, OrderDto orderDto) throws IOException;
    public List<OrderItemDto> getOrderItems(Long orderId) throws Exception ;
    public List<OrderDto> searchOrders(String keyword) ;

}
